src/components/test/PasswordInputField.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { screen } from "@testing-library/react";
+import { renderWithContext, withFormik } from "common";
+import * as Formik from "formik";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import PasswordInputField from "../form/PasswordInputField";
+
+describe("<PasswordInputField>", () => {
+  jest.spyOn(Formik, "useFormikContext").mockImplementation(
+    () =>
+      ({
+        isValid: false,
+        setFieldTouched: jest.fn(),
+        handleSubmit: jest.fn(),
+      } as any)
+  );
+
+  const renderComponent = (propOverrides = {}) => {
+    const props = {
+      externalMaskPassword: undefined,
+      handlePressEndAdornment: undefined,
+      name: "password",
+      showEndAdornment: undefined,
+      ...propOverrides,
+    };
+
+    return renderWithContext(
+      withFormik(<PasswordInputField { ...props } />, {
+        initialValues: { name: "" },
+        onSubmit: jest.fn(),
+      })
+    );
+  };
+
+  it("renders with defaults when only required name prop is given", () => {
+    renderComponent();
+
+    expect(screen.getByTestId("passwordInput")).toBeInTheDocument();
+    expect(screen.getByTestId("VisibilityIcon")).toBeInTheDocument();
```

Comment:
The React Testing libary [docs](https://testing-library.com/docs/queries/about/#priority) recommend using `getByTestId` only when another query won't work, since it doesn't match what the user sees. I'd recommend using the `getByPlaceholderText` query

Reply:
good call, i can add a placeholder prop to it and use that